workflow/common/util.go

```diff
@@ -290,9 +314,58 @@ func ProcessArgs(tmpl *wfv1.Template, args wfv1.Arguments, globalParams, localPa
 	}
 	tmpl.Inputs.Artifacts = newInputArtifacts
 
+	//copy passthrough parameters from Inputs to Arguments
+	if tmpl.Steps != nil {
+		for i, stepList := range tmpl.Steps {
+			for j, step := range stepList {
+				updatedParameters, err := handlePassthroughParameters(tmpl.Inputs, step.Arguments.Parameters)
+				if err != nil {
+					return nil, err
+				}
+				tmpl.Steps[i][j].Arguments.Parameters = updatedParameters
+			}
+		}
+	}
+	if tmpl.DAG != nil {
+		for i, task := range tmpl.DAG.Tasks {
+			updatedParameters, err := handlePassthroughParameters(tmpl.Inputs, task.Arguments.Parameters)
+			if err != nil {
+				return nil, err
+			}
+			tmpl.DAG.Tasks[i].Arguments.Parameters = updatedParameters
+		}
+	}
+
 	return substituteParams(tmpl, globalParams, localParams)
 }
 
+// handlePassthroughParameters returns parameter argument list where items matching PassthroughRegexp are added from input parameters
+func handlePassthroughParameters(inputs wfv1.Inputs, argumentParameters []wfv1.Parameter) ([]wfv1.Parameter, error) {
+	updatedArgumentParameters := make([]wfv1.Parameter, 0)
+
+	// if the input still contains passthrough parameters (eg from FakeArguments.GetParametersByRegexp)
+	// then we can't resolve any passthrough parameters here
+	inputsContainsRegexp := inputs.DoesContainRegexpParameters()
```

Comment:
The comment is not updated. However, same as https://github.com/argoproj/argo/pull/1490/files#r308056868.